perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_associatedUsersMails.java

```diff
@@ -0,0 +1,60 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributesModuleAbstract;
+
+import java.text.Collator;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Module for virtual user attribute associatedUsersMails.
+ * Get list of preferred mails of all associated users.
+ *
+ * @author Radoslav Čerhák <r.cerhak@gmail.com>
+ */
+public class urn_perun_user_attribute_def_virt_associatedUsersMails extends UserVirtualAttributesModuleAbstract {
+
+	private static final String A_U_D_preferredMail = AttributesManager.NS_USER_ATTR_DEF + ":preferredMail";
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) {
+		Attribute attribute = new Attribute(attributeDefinition);
+
+		List<User> users = sess.getPerunBl().getUsersManagerBl().getUsersBySpecificUser(sess, user);
+		List<String> mails = new ArrayList<>();
+
+		try {
+			Attribute userPreferredMail;
+			for (User u : users) {
```

Comment:
We would like to prevent 'one letter' variables. Please change 'u' to 'user'.

Reply:
Fixed